### PR TITLE
Stop spamming logs when setting duplicate receiver

### PR DESCRIPTION
**Pull request by cedric-cordenier**

### CODE REVIEW

File: core/capabilities/syncer.go [hunk 1]
```diff
@@ -543,7 +543,12 @@ func (s *registrySyncer) addReceiver(ctx context.Context, capability kcr.Capabil
 
 	s.lggr.Debugw("Enabling external access for capability", "id", fullCapID, "donID", don.Id)
 	err = s.dispatcher.SetReceiver(fullCapID, fmt.Sprint(don.Id), receiver)
-	if err != nil {
+	if errors.Is(err, remote.ErrReceiverExists) {
+		// If a receiver already exists, let's log the error for debug purposes, but
+		// otherwise short-circuit here. We've handled this capability in a previous iteration.
+		s.lggr.Debugf("failed to set receiver for cap ID %s and don ID %d: %s", fullCapID, don.Id, err)
```
Comment (bolekk) [inline core/capabilities/syncer.go]: Maybe "receiver already exists" instead of "failed to set"?

File: core/capabilities/syncer.go [hunk 2]
```diff
@@ -543,7 +543,12 @@ func (s *registrySyncer) addReceiver(ctx context.Context, capability kcr.Capabil
 
 	s.lggr.Debugw("Enabling external access for capability", "id", fullCapID, "donID", don.Id)
 	err = s.dispatcher.SetReceiver(fullCapID, fmt.Sprint(don.Id), receiver)
-	if err != nil {
+	if errors.Is(err, remote.ErrReceiverExists) {
```
Comment (bolekk) [inline core/capabilities/syncer.go]: Does "Is" work correctly even if it's a longer string with ErrReceiverExists inside it?

Comment (cedric-cordenier) [reply core/capabilities/syncer.go]: It does because we're wrapping the error using the %w directive. The test covers this behaviour

**Review by bolekk — DISMISSED** [review]

File: core/capabilities/remote/dispatcher.go [hunk 3]
```diff
@@ -94,7 +99,7 @@ func (d *dispatcher) SetReceiver(capabilityId string, donId string, rec remotety
 	k := key{capabilityId, donId}
 	_, ok := d.receivers[k]
 	if ok {
-		return fmt.Errorf("receiver already exists for capability %s and don %s", capabilityId, donId)
+		return fmt.Errorf("%w: capability %s and don %s", ErrReceiverExists, capabilityId, donId)
```
Comment (ettec) [inline core/capabilities/remote/dispatcher.go]: nit: I would expect the caller to wrap/log this error with the capability ID etc (as it is doing), so no need to wrap the error at this point.

**Review by ettec — DISMISSED** [review]

**Review by bolekk — DISMISSED** [review]

**Review by krehermann — DISMISSED** [review]

**Pull request closed without merging**